src/policy/mempool_fees.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2024 The Bitcoin Core developers
+// Distributed under the MIT software license. See the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <logging.h>
+#include <node/miner.h>
+#include <policy/estimator.h>
+#include <policy/mempool_fees.h>
+#include <policy/policy.h>
+#include <util/trace.h>
+
+
+using node::GetNextBlockFeeRateAndVsize;
+
+ForecastResult MemPoolPolicyEstimator::EstimateFee(unsigned int targetBlocks)
+{
+    return EstimateFeeWithMemPool(targetBlocks);
+}
+
+ForecastResult MemPoolPolicyEstimator::EstimateFeeWithMemPool(unsigned int targetBlocks)
+{
+    if (targetBlocks > MEMPOOL_FORECAST_MAX_TARGET) {
```

Comment:
In  [policy, fees]: create a mempool based fee rate forecaster:

We don't check that it's >0 here either, so currently `0` is valid.

Reply:
Added an early check for this in the RPC...

Also each forecaster should check for this, because we may call estimatefee via interfaces